content/js1/sprints/2/prep/index.md

```diff
@@ -8,4 +8,406 @@ backlog= 'Module-JS1'
+We need to specify the value want to output from our function. If we don't state what our function should output then the default output will be `undefined`.
+
+### Setting an output
+
+To specify a function's output, we write a `return` statement.
+
+If your function call is like a question, the return statement is the answer. It's what comes back.
+
+We can add a return statement to the function as follows:
+
+```js {linenos=table,hl_lines=["4"],linenostart=1}
+function convertToPercentage() {
+  const num = 0.5;
+  const percentage = `${num * 100}%`;
+  return percentage;
+}
+```
+
+Now that our function contains a `return` statement we can re-run our script to check the output.
+
+### Checking with different inputs
+
+Now we have a function declaration we can check if our function works for any number. We can call `convertToPercentage` with different arguments and check the outputs we get each time.
+
+```js {linenos=table,hl_lines=["7-8"],linenostart=1}
+function convertToPercentage() {
+  const num = 0.5;
+  const percentage = `${num * 100}%`;
+  return percentage;
+}
+
+const output1 = convertToPercentage(0.5);
+const output2 = convertToPercentage(0.3);
+```
+
+When we execute the code we _want_ to log the target output for each input `0.5` and `0.3`:
+
+```bash
+50%
+23%
+```
+
+However, _given the function's current implementation_, we get the following logs:
+
+```bash
+50%
+50%
+```
+
+### 💻 Playing computer
+
+To understand how our function currently behaves we must start building an understanding of how our code is executed. We might currently have some ideas of how code is executed in our program:
+
+```mermaid
+mindmap
+Root[How is a piece of code executed?]
+    B[Lines of code are executed from top to bottom]
+    C[Variable declarations mean data is stored in memory]
+    D[Function call means running the instructions inside the function]
+    E[When returning from a function we get an output]
+```
+
+### ➡️ Current line of code
+
+We can make use of execution diagrams to visualise how a program is executed by a computer.
+
+Firstly, we need a way to track the **line of code that is currently executing.** In the diagrams below, we use an arrow to indicate which line of code is currently being executed. Each line is an instruction to the computer that will produce some effect. For example, a line of code with a variable declaration means "store a new variable with a particular value".
+
+Let's consider an example:
+
+<iframe width="800" height="300" frameborder="0" src="https://pythontutor.com/iframe-embed.html#code=const%20name%20%3D%20'Gayle'%3B&codeDivHeight=400&codeDivWidth=350&cumulative=false&curInstr=0&heapPrimitives=nevernest&origin=opt-frontend.js&py=js&rawInputLstJSON=%5B%5D&textReferences=false"> </iframe>
+
+In the example above, we have a single variable declaration:
+
+```js
+const name = "Gayle";
+```
+
+which means store a string `"Gayle"` in memory.
+
+### 🖼️ Storing things in memory
```

Comment:
I think before introducing the idea of frames, it would be useful to show an example of what a trainee may think would work that doesn't, which motivates the understanding?

We could show an example of `console.log(percentage)` after the two outputs above, and have them guess what it prints? Then go into this explanation by way of explaining why it doesn't work like they'd expect?